File: bank-core/src/main/kotlin/com/example/domain/models/Money.kt
package com.example.domain.models

import com.example.data.entities.CurrencyType
import com.example.domain.utils.BigDecimalSerializer
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.math.BigDecimal

data class Money(
    val amount: BigDecimal,
    val currencyType: CurrencyType
) {
    fun convertTo(targetCurrency: CurrencyType, exchangeRate: BigDecimal): Money {
        if (currencyType == targetCurrency) {
            return this
        }
        val convertedAmount = amount.multiply(exchangeRate)
        return Money(convertedAmount, targetCurrency)
    }
}

File: WebApplication1/CreditService.DAL/Entities/Money.cs
using CreditService.DAL.Enum;

namespace CreditService.DAL.Entities;


public class Money
{
    public decimal Amount { get; private set; }
    public CurrencyType Currency { get; private set; }
    
    private Money() { } 

    public Money(decimal amount, CurrencyType currency)
    {
        Amount = amount;
        Currency = currency;
    }
    
}